### PR TITLE
Deployed authentication. Add getByLogin method in persistence, service and api layers.

### PR DESCRIPTION
**Pull request by VolodymyrPortianko**

### CODE REVIEW

File: api/src/main/java/com/bionicuniversity/edu/fashiontips/api/AuthController.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+import org.springframework.web.bind.annotation.*;
+
+/**
+ *  Controller which handles authenticate requests
+ */
+@CrossOrigin
+@RestController
+public class AuthController {
+
+    @RequestMapping(value = "/me", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
```
Comment (mxfmd) [inline api/src/main/java/com/bionicuniversity/edu/fashiontips/api/AuthController.java:17]: How about this implementation (with java.security.Principal instance)?

```
@RequestMapping(value = "/me", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
public ResponseEntity<String> login(Principal principal) {
    return new ResponseEntity<>(principal.getName(), HttpStatus.OK);
}
```

Comment (mxfmd) [inline api/src/main/java/com/bionicuniversity/edu/fashiontips/api/AuthController.java:17]: Why do you need this peace of code?

MediaType.TEXT_PLAIN_VALUE

Can it works without explicitly set media type?

Comment (VolodymyrPortianko) [inline api/src/main/java/com/bionicuniversity/edu/fashiontips/api/AuthController.java:17]: @RestController by default sends Header - MediaType.Json.
This method don't return Json, just simple string, so we must set it manually.

Comment (VolodymyrPortianko) [inline api/src/main/java/com/bionicuniversity/edu/fashiontips/api/AuthController.java:17]: This works too.

```
@RequestMapping(value = "/me", method = RequestMethod.GET, produces = MediaType.TEXT_PLAIN_VALUE)
public ResponseEntity<String> login(Principal principal) {
   return new ResponseEntity<>(principal.getName(), HttpStatus.OK);
   }
```

**Pull request merged**